install/kubernetes/cilium/charts/config/templates/configmap.yaml

```diff
@@ -611,3 +611,8 @@ data:
 {{- else if ( ne $crdWaitTimeout "5m" ) }}
   crd-wait-timeout: {{ $crdWaitTimeout | quote }}
 {{- end }}
+
+{{- if and .Values.global.k8s .Values.global.k8s.serviceProxyName }}
```

Comment:
Taking a closer look I see that we use `hasKey` only for local configmap sub chart values and not the global values. I am using a similar approach already used in the configmap - https://github.com/cilium/cilium/blob/master/install/kubernetes/cilium/charts/config/templates/configmap.yaml#L503

But yeah, it would be good to know/have better standards that we are aiming for Cilium. As @seanmwinn mentioned in the Ad-hoc meeting regarding helm options.